File: src/main/java/com/goudourasv/data/lectures/LecturesMapper.java
package com.goudourasv.data.lectures;

import com.goudourasv.data.courses.CourseEntity;
import com.goudourasv.domain.courses.CourseData;
import com.goudourasv.domain.lectures.Lecture;

import java.util.ArrayList;
import java.util.List;

public class LecturesMapper {
    public static Lecture toLecture(LectureEntity lectureEntity, CourseEntity courseEntity) {
        CourseData course = new CourseData(courseEntity.getId(), courseEntity.getTitle());
        Lecture lecture = new Lecture(lectureEntity.getId(), lectureEntity.getTitle(), lectureEntity.getUri(), course, lectureEntity.getStartTime(), lectureEntity.getEndTime());
        return lecture;
    }

    public static List<Lecture> toLectures(List<LectureEntity> lectureEntities) {
        List<Lecture> lectures = new ArrayList<>();
        for (LectureEntity lectureEntity : lectureEntities) {
            CourseEntity courseEntity = lectureEntity.getCourseEntity();
           Lecture lecture = toLecture(lectureEntity,courseEntity);
            lectures.add(lecture);
        }
        return lectures;
    }
    public static List<LectureEntity> toLectureEntities(List<Lecture> lectures){
        List<LectureEntity> lectureEntities = new ArrayList<>();
        for (Lecture lecture : lectures) {
            LectureEntity lectureEntity = new LectureEntity(lecture.getId(), lecture.getTitle(), lecture.getUri(), lecture.getStartTime(), lecture.getEndTime());
            lectureEntities.add(lectureEntity);
        }
        return lectureEntities;
    }
}

File: src/main/java/com/goudourasv/data/lectures/LecturesMapper.kt
package com.goudourasv.data.lectures

import com.goudourasv.data.courses.toCourse
import com.goudourasv.data.courses.toCourseData
import com.goudourasv.data.courses.toCourseEntity
import com.goudourasv.domain.lectures.Lecture
import com.goudourasv.domain.lectures.LectureData


fun LectureEntity.toLecture(): Lecture {
    return Lecture(
        id = this.id!!,
        title = this.title,
        uri = this.uri,
        startTime = this.startTime,
        endTime = this.endTime,
        course = this.courseEntity!!.toCourse(),
    )
}

fun List<LectureEntity>.toLectures(): List<Lecture> {
    return this.map { it.toLecture() }
}

fun List<Lecture>.toLectureEntities(): List<LectureEntity> {
    return this.map { it.toLectureEntity() }

}

fun Lecture.toLectureEntity(): LectureEntity{
    return LectureEntity(
        id = this.id,
        title = this.title,
        uri = this.uri,
        startTime = this.startTime,
        endTime = this.endTime,
        courseEntity = this.course.toCourseEntity(),
    )
}

fun LectureData.toLectureEntity(): LectureEntity {
    return LectureEntity(
        id = this.id,
        title = this.title,
        uri = this.uri,
        startTime = this.startTime,
        endTime = this.endTime,
        courseEntity = null,
    )
}

fun LectureEntity.toLectureData(): LectureData{
    return LectureData(
        id = this.id!!,
        title = this.title,
        uri = this.uri,
        startTime = this.startTime,
        endTime = this.endTime,
    )
}
fun Lecture.toLectureData() : LectureData {
    return  LectureData(
        id = this.id,
        title = this.title,
        uri = this.uri,
        startTime = this.startTime,
        endTime = this.endTime,
    )
}
